src/main/java/com/ldtteam/structures/blueprints/v1/Blueprint.java

```diff
@@ -230,7 +231,7 @@ public void addBlockState(final BlockPos pos, final BlockState state)
 
         if (index == -1)
         {
-            index = this.palleteSize + 1;
+            index = this.palleteSize;
```

Comment:
how is this bug when it would affect the blueprints immediately? I don't see any how this would not break the game already since it affects the raw saved data

Reply:
Shape tool blueprints weren't saved (before).  Scan tool blueprints use the other constructor, which calculated it correctly.

The shape tool preview didn't crash because there's the reverse of this bug down in `addBlockState` that caused it to calculate the correct index (based on the wrong initial state).  However the `palletteSize` was always too small, so if you actually tried to save that blueprint it would miss one block from the palette and would crash whoever tried to load it.

TBH the whole `palleteSize` field seems entirely unnecessary (and only a potential source of bugs).  It'd be better to get rid of it entirely, but I didn't want to make too many changes at once.